File: 2_modules/geom.erl
-module(geom).
-export([area/2]).
-include_lib("eunit/include/eunit.hrl").

area(Width,Height) -> Width * Height.

area_test_() -> [
  ?_assert( area(3,4)   =:= 12  ),
  ?_assert( area(0,4)   =:= 0   ),
  ?_assert( area(1.5,4) =:= 6.0 )
].

File: ex03_pattern_matching/lib/geom.ex
defmodule Geom do
  @moduledoc """
    Calculates the area of geometric shapes.
    """

  @doc "area(shape, width, height) -> the area of a rectange with the given width and height."
  def area(:rectangle, width, height) do
    (width * height)
  end

  def area(:triangle, base, height) do
    (base * height)/2.0
  end

  def area(:ellipse, major, minor) do
    :math.pi() * major * minor
  end
end
